File: DynamicKey/AgoraDynamicKey/nodejs/src/FpaTokenBuilder.js
const AccessToken = require('../src/AccessToken2').AccessToken2
const ServiceFpa = require('../src/AccessToken2').ServiceFpa

class FpaTokenBuilder {
    /**
     * Build the FPA token.
     * @param appId The App ID issued to you by Agora. Apply for a new App ID from
     * Agora Dashboard if it is missing from your kit. See Get an App ID.
     * @param appCertificate Certificate of the application that you registered in
     * the Agora Dashboard. See Get an App Certificate.
     * @return The FPA token.
     */
    static buildToken(appId, appCertificate) {
        let token = new AccessToken(appId, appCertificate, 0, 24 * 3600)

        let serviceFpa = new ServiceFpa()
        serviceFpa.add_privilege(ServiceFpa.kPrivilegeLogin, 0)
        token.add_service(serviceFpa)

        return token.build()
    }
}

module.exports.FpaTokenBuilder = FpaTokenBuilder

File: DynamicKey/AgoraDynamicKey/dart/lib/src/fpa_token_builder.dart
import 'access_token.dart';

/// Builds Token007 tokens for FPA login.
class FpaTokenBuilder {
  static const int _tokenExpireSeconds = 24 * 60 * 60;

  /// Builds an FPA token.
  ///
  /// [appId] is the App ID issued by Agora; create one in Agora Dashboard if needed.
  /// [appCertificate] is the certificate of the application registered in Agora Dashboard.
  ///
  /// Returns the generated FPA token.
  static String buildToken({
    required String appId,
    required String appCertificate,
  }) {
    final token = AccessToken.create(
      appId,
      appCertificate,
      expire: _tokenExpireSeconds,
    );
    token.addService(
      ServiceFpa()..addPrivilege(ServiceFpa.privilegeLogin, 0),
    );
    return token.build();
  }
}
